### PR TITLE
Add Vertical Navigation with Persistent Secondary

### PR DESCRIPTION
**Pull request by jeff-phillips-18**

### CODE REVIEW

File: tests-src/vertical-navigation.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  $(function() {
+    // matchHeight the contents of each .card-pf and then the .card-pf itself
+    $(".row-cards-pf > [class*='col'] > .card-pf .card-pf-title").matchHeight();
+    $(".row-cards-pf > [class*='col'] > .card-pf > .card-pf-body").matchHeight();
+    $(".row-cards-pf > [class*='col'] > .card-pf > .card-pf-footer").matchHeight();
+    $(".row-cards-pf > [class*='col'] > .card-pf").matchHeight();
+    // initialize tooltips
+    $('[data-toggle="tooltip"]').tooltip();
+  });
+  $(document).ready(function() {
```
Comment (dtaylor113) [inline tests-src/vertical-navigation.html]: Should all of this be in the $(document).ready(function() function?  Why the self-executing and the doc.ready() ?

**Comment by dtaylor113** [comment]
Cloned and reviewed test pages.  LGTM, although I don't like the way the menu title becomes bold and wider when hovering over them -but if that is as designed...

**Comment by jeff-phillips-18** [comment]
@andresgalante Could you please review? Thanks!

File: less/layouts.less [hunk 2]
```diff
@@ -2,6 +2,81 @@
...
+    min-height: 100%
+  }
+  &.layout-pf-fixed {
+    &.transitions .container-pf-nav-pf-vertical {
+      transition: @flyout-transition-pf;
+    }
+    body {
+      padding-top: @navbar-pf-height; // make space for the navbar
+    }
+
```
Comment (andresgalante) [inline less/layouts.less]: @jeff-phillips-18 I know this is silly, but we don't usually leave spaces between declarations

**Comment by andresgalante** [comment]
@jeff-phillips-18 we are missing `-pf` postfix on all our classes

**Comment by andresgalante** [comment]
@jeff-phillips-18 heighlighting of items is wrong:
screen shot 2016-02-22 at 1 29 36 pm

File: less/vertical-nav.less [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+      transition-delay: .15s;
+    }
+  }
+}
+
+.nav-pf-vertical {
+  background: @nav-pf-vertical-bg-color;
+  border-color: @nav-pf-vertical-border-color;
+  border-style: solid;
+  border-width: 0 1px 0 0;
```
Comment (andresgalante) [inline less/vertical-nav.less]: @jeff-phillips-18 you can replace this with `border-right: 1px solid @color`

File: less/vertical-nav.less [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+    width: @nav-pf-vertical-collapsed-width;
+    > .list-group {
+      > .list-group-item {
+
+        > a {
+          margin-right: 0;
+          width: @nav-pf-vertical-collapsed-width;
+
+          > .list-group-item-value {
+            width: 0;
```
Comment (andresgalante) [inline less/vertical-nav.less]: alpha

File: less/vertical-nav.less [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+  .layout-pf-fixed-with-footer & {
+    bottom: @footer-pf-height;
+  }
+
+  > .list-group {
+    border-top: 0;
+    margin-bottom: 0;
+
+    > .list-group-item {
+      padding: 0;
```
Comment (andresgalante) [inline less/vertical-nav.less]: alpha

File: less/vertical-nav.less [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+      background-color: transparent;
+      border-color: @nav-pf-vertical-item-border-color;
+
+      > a {
+        width: @nav-pf-vertical-width;
```
Comment (andresgalante) [inline less/vertical-nav.less]: alpha

File: less/vertical-nav.less [hunk 7]
```diff
@@ -0,0 +1,467 @@
...
+        padding: @nav-pf-vertical-link-padding;
+        position: relative;
+        white-space: nowrap;
+
+        .fa,
+        .glyphicon,
+        .pficon {
+          color: @nav-pf-vertical-icon-color;
+          float: left;
+          font-size: 24px;
```
Comment (andresgalante) [inline less/vertical-nav.less]: 24px is `(@font-size-base*2)` since font size base is 14

File: less/vertical-nav.less [hunk 8]
```diff
@@ -0,0 +1,467 @@
...
+
+        &:hover, &:focus {
+          text-decoration: none;
+        }
+      }
+
+      &.active, &:hover {
+        > a {
+          color: @nav-pf-vertical-active-color;
+          background-color: @nav-pf-vertical-active-bg-color;
```
Comment (andresgalante) [inline less/vertical-nav.less:205]: alpha

File: less/vertical-nav.less [hunk 9]
```diff
@@ -0,0 +1,467 @@
...
+        width: 140px;
+      }
+
+      &.persistent-secondary {
+        > a:after {
+          content: "\f105";
+          color: @nav-pf-vertical-secondary-indicator-color;
+          display: block;
+          font-family: "FontAwesome";
+          font-size: 24px;
```
Comment (andresgalante) [inline less/vertical-nav.less]: also here, you can use font size base times 2

File: less/vertical-nav.less [hunk 10]
```diff
@@ -0,0 +1,467 @@
...
+      display: none;
+    }
+
+    .show-mobile-nav & {
+      left: 0;
+      z-index: (@zindex-navbar-fixed + 2);
+    }
+  }
+
+  .persistent-secondary.hovered-nav-item & {
```
Comment (andresgalante) [inline less/vertical-nav.less]: can this be `:hover` ?

File: less/vertical-nav.less [hunk 11]
```diff
@@ -0,0 +1,467 @@
...
+    display: block;
+    z-index: (@zindex-navbar-fixed + 1);
+    .collapsed & {
+      left: @nav-pf-vertical-collapsed-width;
+    }
+    .show-mobile-nav & {
```
Comment (andresgalante) [inline less/vertical-nav.less]: to keep consistency with bootstrap names we use `visible` and `hidden`

File: tests-src/vertical-navigation.html (hunk 1 above)
Comment (jeff-phillips-18) [inline tests-src/vertical-navigation.html]: Fixed

File: less/layouts.less (hunk 2 above)
Comment (jeff-phillips-18) [inline less/layouts.less]: Fixed

**Comment by jeff-phillips-18** [comment]
I believe I have addressed all the issues above.

**Comment by dtaylor113** [comment]
Hi @jeff-phillips-18 @andresgalante ,
Is this good to go?
Thanks,
- Dave

**Comment by jeff-phillips-18** [comment]
Added toast notifications to provide the standard "These examples are included for development testing purposes. " blurb on the full page tests.

**Comment by jeff-phillips-18** [comment]
Fixed issue with horizontal scrollbar appearing when secondary menu is shown.

File: less/layouts.less [hunk 12]
```diff
@@ -1,6 +1,82 @@
...
+    .container-pf-nav-pf-vertical {
+      margin-left: @nav-pf-vertical-width;
+
```
Comment (andresgalante) [inline less/layouts.less]: remove this space

File: less/layouts.less [hunk 13]
```diff
@@ -1,6 +1,82 @@
...
+      &.hidden-icons-pf {
+        margin-left: @nav-pf-vertical-hidden-icons-width;
+      }
+      &.collapsed-nav {
+        margin-left: 0; // adjust space for the collapsed left nav
+      }
+      &.hidden-nav {
+        margin-left: 0; // remove space as left nav is hidden
+      }
+
```
Comment (andresgalante) [inline less/layouts.less]: And this one, and any other blank line. we just don't leave blank lines on patternfly. Thats the way its done so far so lets keep consistency.

File: tests-src/_includes/widgets/secondary-nav-storage.html [hunk 14]
```diff
@@ -0,0 +1,126 @@
+<div id="storage-secondary" class="nav-pf-persistent-secondary">
+  <div class="persistent-secondary-header">
+    <a class="fa fa-arrow-circle-o-left" href="#" data-toggle="collapse-secondary-nav"></a>
+    <span>Storage</span>
+  </div>
+
```
Comment (andresgalante) [inline tests-src/_includes/widgets/secondary-nav-storage.html]: please remove this space

File: tests-src/_includes/widgets/secondary-nav-storage.html [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+  <h5 class="list-group-header">File</h5>
+  <ul class="list-group">
+    <li class="list-group-item active">
+      <a href="#">
+        <span class="list-group-item-value">Volumes</span>
+        <div class="badge-container">
```
Comment (andresgalante) [inline tests-src/_includes/widgets/secondary-nav-storage.html]: @jeff-phillips-18 if this class is ours then we should add a -pf. Now if thats something from the alt layout then keep it this way

File: tests-src/_includes/widgets/secondary-nav-storage.html [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+          <div class="badge">
+            <span class="pficon pficon-error-circle-o"></span>
+            <span>1</span>
+          </div>
+          <span class="badge">10</span>
+        </div>
+      </a>
+    </li>
+  </ul>
+  <h5 class="list-group-header">Object</h5>
```
Comment (andresgalante) [inline tests-src/_includes/widgets/secondary-nav-storage.html]: @jeff-phillips-18 I think we've talked about this one. I don't remember. so if this is from the alt layout lets keep it, if not lets add the -pf. Or just target it on the css with `nav-pf-persistent-secondary h5` I don't think there will be other h5s on this vertical menu. whatever you think is best.

File: tests-src/_includes/widgets/vertical-nav-with-secondary.html [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+        <span class="list-group-item-value">Clusters</span>
+      </a>
+    </li>
+    <li class="list-group-item">
+      <a href="#">
+        <span class="pficon pficon-screen"></span>
+        <span class="list-group-item-value">Hosts</span>
+      </a>
+    </li>
+    <li class="list-group-item persistent-secondary" data-target="#storage-secondary">
```
Comment (andresgalante) [inline tests-src/_includes/widgets/vertical-nav-with-secondary.html:26]: if `persistent-secondary` is ours, it needs a `-pf`

File: less/vertical-nav.less [hunk 18]
```diff
@@ -0,0 +1,487 @@
...
+
+        &.show-mobile-nav {
+          width: @nav-pf-vertical-secondary-width;
+        }
+      }
+    }
+  }
+
+  &.hidden {
+    display: none;
```
Comment (andresgalante) [inline less/vertical-nav.less]: hidden by default show display none. do you need it?

File: less/vertical-nav.less [hunk 19]
```diff
@@ -0,0 +1,487 @@
...
+
+      &.collapsed {
+        width: @nav-pf-vertical-collapsed-width;
+      }
+
+      &.show-mobile-nav {
+        width: @nav-pf-vertical-secondary-width;
+      }
+
+      @media (min-width: 1200px) {
```
Comment (andresgalante) [inline less/vertical-nav.less]: @jeff-phillips-18 change it for @media (min-width: @screen-lg-min) { } lg is 1200px

File: tests-src/_includes/widgets/vertical-nav-with-secondary.html (hunk 17 above)
Comment (jeff-phillips-18) [inline tests-src/_includes/widgets/vertical-nav-with-secondary.html:26]: This was introduced by the alt layouts :(

Comment (andresgalante) [inline tests-src/_includes/widgets/vertical-nav-with-secondary.html:26]: @jeff-phillips-18 dont worry, just ignore my comments when its alt layouts fault :)

File: less/layouts.less (hunk 12 above)
Comment (jeff-phillips-18) [inline less/layouts.less]: done

File: less/layouts.less (hunk 13 above)
Comment (jeff-phillips-18) [inline less/layouts.less]: done

File: tests-src/_includes/widgets/secondary-nav-storage.html (hunk 14 above)
Comment (jeff-phillips-18) [inline tests-src/_includes/widgets/secondary-nav-storage.html]: done

File: tests-src/_includes/widgets/secondary-nav-storage.html (hunk 15 above)
Comment (jeff-phillips-18) [inline tests-src/_includes/widgets/secondary-nav-storage.html]: done

File: tests-src/_includes/widgets/secondary-nav-storage.html (hunk 16 above)
Comment (jeff-phillips-18) [inline tests-src/_includes/widgets/secondary-nav-storage.html]: done, I targeted the h5

File: less/vertical-nav.less (hunk 18 above)
Comment (jeff-phillips-18) [inline less/vertical-nav.less]: Nope, removed.

File: less/vertical-nav.less (hunk 19 above)
Comment (jeff-phillips-18) [inline less/vertical-nav.less]: Done.

**Comment by jeff-phillips-18** [comment]
I have address all the issues above.

**Comment by andresgalante** [comment]
@jeff-phillips-18 Excellent, ship it! :rocket:

**Pull request merged**